src/main/java/dao/UserDao.java

```diff
@@ -1,4 +1,74 @@
 package dao;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import model.User;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
 public class UserDao {
+    public static String usersJsonFile = "users.json";
+
+    public static List<User> users = new ArrayList<User>() {{
+        add(new User(1, "Ben", true));
+        add(new User(2, "Alice", true));
+        add(new User(3, "Bill", false));
+        add(new User(4, "John", false));
+    }};
+
+    public static void blockUser(int userId, boolean block) {
+        List<User> users = getUsers();
+        users.stream()
+                .filter(v -> v.getId() == userId)
+                .forEach(v -> v.blockUser(block));
+        writeJsonToFile(users, usersJsonFile);
+    }
+
+    public static List<User> getUsers() {
+        return readJsonFile(usersJsonFile);
+    }
+
+    public static List<User> getBlockedUsersList(boolean block) {
+        return getUsers().stream()
+                .filter(v -> v.isBlocked() == block)
+                .collect(Collectors.toList());
+    }
+
+    public static void writeJsonToFile(List<User> jsonString, String jsonFile) {
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
why do store it  json format?

Reply:
I thought it was better than storing data in variables, and easier (for now) than storing data in a database.

Reply:
Changed storing data from JSON file to in memory.